app/services/teamdynamix_api.rb

```diff
@@ -97,51 +96,23 @@ def parse_response(res)
     end
   end
 
-  def valid_auth_token?(token)
-    token.match(/^[a-zA-Z0-9\.\-\_]*$/)
-  end
-
   def retire_asset_payload(asset_id)
     asset = get_asset(asset_id)
     asset.merge!(API_CONFIG[:delete].stringify_keys)
   end
 
   def create_asset_payload(host)
+    @host = host
     ensure_configured_create_params
     payload = { AppID: APP_ID,
                 SerialNumber: host.name,
                 Name: host.fqdn }
-    payload.merge!(API_CONFIG[:create].stringify_keys)
-    payload.merge(Attributes: create_asset_attributes(host))
-  end
-
-  def create_asset_attributes(host)
-    [
-      {
-        ID: 11_632,
-        Name: 'mu.ci.Description',
-        Value: "Foreman host #{host.fqdn} created by ForemanTeamdynamix plugin"
-      },
-      {
-        ID: 11_634,
-        Name: 'mu.ci.Lifecycle Status',
-        Value: lifecycle_status
-      }
-    ]
-  end
-
-  def lifecycle_status
-    case Rails.env.downcase
-    when 'test' then 26_190
-    when 'development' then 26_191
-    when 'stage', 'pre-production' then 26_192
-    when 'early-life-support' then 26_194
-    when 'production' then 26_193
-    end
-  end
 
-  def must_configure_create_params
-    [:StatusID]
+    config_attrs = API_CONFIG[:create].symbolize_keys
+    custom_attrs = config_attrs[:CustomAttributes]
+    payload[:Attributes] = custom_attr_fields(host, custom_attrs) if custom_attrs.present?
+    config_attrs.delete(:CustomAttributes)
```

Comment:
delete returns the value, so you can remove this line and use this for line 112
custom_attrs = config_attrs.delete(:CustomAttributes)

Reply:
makes sense